.github/workflows/relay.yml

```diff
@@ -42,7 +42,17 @@ jobs:
         run: nix develop -c go build -v ./pkg/cosmos/...
 
       - name: Run unit tests
-        run: nix develop -c go test -v ./pkg/cosmos/...
+        run: nix develop -c go test -v -covermode=atomic ./pkg/cosmos/... -coverpkg=./... -coverprofile=coverage.txt
 
       - name: Run tests with the race detector enabled
-        run: nix develop -c go test -v ./pkg/cosmos/... -race -count=10
+        run: nix develop -c go test -v ./pkg/cosmos/... -race -count=10 -coverpkg=./... -coverprofile=race_coverage.txt
```

Comment:
Declaring custom CI steps inside a workflow is deprecated.

Please move these actions to Makefile and trigger them with make - we want all relevant project actions in a single place.